lib/src/repositories/connect_bluetooth_device_repository.dart

```diff
@@ -77,18 +83,16 @@ class ConnectBluetoothDeviceRepository {
 
   void _successfullyConnected(BluetoothDevice device) {
     _connectionStateSubscription = device.connectionState.listen((state) {
+      debugPrint('device connection state: $state');
       bluetoothConnectionState = state;
-      if (state == BluetoothConnectionState.disconnected) {
-        _device = null;
-      }
```

Comment:
shouldn't do this if a disconnect can happen, want to keep the instance of the `BluetoothDevice` we've connected with/paired to